tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -0,0 +1,270 @@
+// <copyright file="Test.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace.Ci.Tagging;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Pdb;
+
+namespace Datadog.Trace.Ci;
+
+/// <summary>
+/// CI Visibility test
+/// </summary>
+public sealed class Test
+{
+    private static readonly AsyncLocal<Test?> CurrentTest = new();
+    private readonly Scope _scope;
+    private int _finished;
+
+    internal Test(TestSuite suite, string name, DateTimeOffset? startDate = null)
```

Comment:
I'd suggest avoiding the default parameter given this is a public class etc. They just tend to make for messier APIs overall IMO (especially as you can't change it due to version mismatch)

Reply:
.ctor are not public, but I did remove the optional parameters from the static methods on [1d3508b](https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/1d3508be7337c2024f64559ad008daeef8f2921f)